### PR TITLE
feat(KlaviyoCore): QueueStore.drainAll + registry seam + load warning (MAGE-952)

### PR DESCRIPTION
**Pull request by belleklaviyo**
# Description

Additive, dormant KlaviyoCore primitives for the MAGE-952 flush repoint, plus two behavior-preserving KlaviyoSwift refactors, split out of #706 so the large KlaviyoSwift change reviews on its own. Nothing consumes the new Core primitives yet — same pattern as MAGE-950/951 shipping their Core pieces ahead of the consumer.

## Due Diligence
- [x] I have tested this on a simulator or a physical device.
- [x] I have added sufficient unit/integration tests of my changes.
- [x] I am confident these changes are compatible with all iOS and XCode versions the SDK currently supports.

## Release/Versioning Considerations
- [x] `Minor` Contains changes to the public API.
- [x] This is planned work for an upcoming release.

Public API delta: adds `QueueStore.drainAll(persist:)`. Adds a `package` `register(_:for:)` test seam. `restore()` unchanged. Replaces `RequestEnqueuer.enqueueProfile(properties:)` with `enqueueProfile(payload:)` (introduced in #703/MAGE-951, unused until the MAGE-952 consumer).

## Changelog / Code Overview

**KlaviyoCore (dormant primitives)**
- `drainAll(persist:)` — atomic snapshot-and-clear of the pending queue (the MAGE-952 flush loop leases through it).
- `register(_:for:)` — package test-injection seam so reducer tests can back the queue with an in-memory spy (`.test` file stubs are no-ops).
- `hydrated()` — emit a developer warning on a `diskIO.load()` failure before the empty fallback, so overwriting a valid-but-unreadable backlog is observable rather than silent.
- `RequestEnqueuer.enqueueProfile(payload:)` — takes a prebuilt `CreateProfilePayload` instead of identity + a flat properties dict, so the KlaviyoSwift consumer can carry a profile's structured attributes (name/title/organization/image/location) through the durable buffer rather than dropping them (unblocks MAGE-1141). No `UnattributedRequest`/on-disk change — the buffered case already stores a full payload.

**KlaviyoSwift (behavior-preserving refactors)**
- Extract `KlaviyoState` request-building helpers into an extension — pure relocation, no behavior change.
- Introduce a dedicated `LegacyState` Codable DTO that owns decoding of the legacy `klaviyo-{apiKey}-state.json` blob, and point `LegacyStateMigration` at it instead of `KlaviyoState`.

### Why `LegacyState`?

The only remaining reason `KlaviyoState` still conforms to `Codable` is that the one-time legacy-file migration decodes that on-disk blob directly into it. In the MAGE-952 flush repoint, persistence is owned by the canonical Core stores (QueueStore, IdentityStore, etc.) and `KlaviyoState` becomes a purely in-memory runtime type — it no longer needs to be serializable.

Moving the legacy blob's decoding into its own DTO lets the consumer PR (#706) drop `Codable` from `KlaviyoState`. Benefits:
- **Decouples the runtime type from the on-disk format.** The migration owns the legacy shape (including the old top-level-identity fallback); `KlaviyoState` is free to evolve without worrying about wire/disk compatibility.
- **Removes accidental persistence coupling.** Once `KlaviyoState` isn't `Codable`, nothing can silently re-serialize it to disk, keeping the canonical Core stores the single source of truth.
- **Isolates legacy quirks.** All the "read the pre-split blob" edge cases live in one migration-owned DTO with its own tests, rather than leaking into the runtime state type.

This change is behavior-preserving on its own — `LegacyState` decodes exactly what `KlaviyoState` did (new nested-identity format, legacy top-level identity, and queue-only blobs) — and simply unblocks the Codable removal on the consumer PR.

Behavior-preserving for existing flows.

## Test Plan
- `KlaviyoCoreTests` green (new `drainAll` tests, `register` test, load-failure warning test).
- `KlaviyoSwiftTests` green (new `LegacyStateTests` covering nested-identity, legacy top-level identity, and queue-only decoding).

## Related Issues/Tickets
- MAGE-952 (parent). Consumer PR: #706 (stacked on this).

## Summary by CodeRabbit

* **New Features**
  * Added the ability to atomically retrieve and clear all queued requests.
  * Added configurable persistence behavior when draining queued requests.
  * Added support for registering queue stores by API key.
  * Added structured request-building support for profiles, push tokens, and subscriptions.
  * Profile requests now preserve structured attributes when buffered or queued.

* **Bug Fixes**
  * Queue loading failures now emit a warning and safely fall back to an empty queue.
  * Improved migration of saved state from older formats, including legacy identity and queue data.
  * Invalid subscription configurations and missing identifiers now prevent invalid requests from being queued.

### CODE REVIEW

File: Sources/KlaviyoSwift/StateManagement/KlaviyoState+RequestBuilding.swift [hunk 1]
```diff
@@ -0,0 +1,197 @@
+//
+//  KlaviyoState+RequestBuilding.swift
```
Comment (belleklaviyo) [inline Sources/KlaviyoSwift/StateManagement/KlaviyoState+RequestBuilding.swift:2]: this file is purely a move of some funcs to a separate file

File: Sources/KlaviyoCore/QueueStore.swift [hunk 2]
```diff
@@ -228,7 +242,14 @@ public final class QueueStore {
     /// Loads from disk on first access; memory is authoritative thereafter. Call under `queueLock`.
     private func hydrated() -> [KlaviyoRequest] {
         if let queue { return queue }
-        let loaded = (try? diskIO.load()) ?? []
+        let loaded: [KlaviyoRequest]
+        do {
+            loaded = try diskIO.load()
+        } catch {
+            // TODO: Distinguish corrupt-vs-transient to preserve the latter
```
Comment (belleklaviyo) [inline Sources/KlaviyoCore/QueueStore.swift]: https://linear.app/klaviyo/issue/MAGE-1142/queuestore-distinguish-corrupt-vs-transiently-unreadable-load-failures

ticketed here but honestly probably dont need to do, can't imagine a corrupt yet also recoverable file... mostly a coderabbit catch

File: Sources/KlaviyoCore/RequestEnqueuer.swift [hunk 3]
```diff
@@ -59,9 +59,13 @@ public enum RequestEnqueuer {
...
 
-    public static func enqueueProfile(properties: [String: Any]) {
-        guard let identity = resolveIdentity() else { return }
-        let payload = RequestFactory.profilePayload(identity: identity, properties: properties)
+    /// Enqueues an already-built `CreateProfilePayload`. Unlike the other entry points, the caller
+    /// supplies the full payload — profiles carry structured attributes (firstName/lastName/title/
+    /// organization/image/location) that only the KlaviyoSwift `Profile` → `ProfilePayload` mapping
+    /// can populate, so building here (with just identity + flat properties) would drop them. The
+    /// payload already embeds identifiers + anonymousId; routing is the same as every other request:
+    /// apiKey present → `QueueStore`, absent → durable `UnattributedBuffer`. (MAGE-1141)
```
Comment (glenn-klaviyo) [inline Sources/KlaviyoCore/RequestEnqueuer.swift]: should we remove the ticket numbers from this PR?

File: Sources/KlaviyoSwift/StateManagement/KlaviyoState+RequestBuilding.swift [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+import AnyCodable
+import Foundation
+import KlaviyoCore
+
+// MARK: - Request-building helpers
+
+extension KlaviyoState {
+    /// Resolves the profile for a token request, folding in and consuming any pending profile.
+    private mutating func resolveProfileConsumingPending() -> Profile {
```
Comment (glenn-klaviyo) [inline Sources/KlaviyoSwift/StateManagement/KlaviyoState+RequestBuilding.swift]: I feel this function name is hard to follow. what about something like `resolveAndConsumePendingProfile()` or `resolveProfile()` 🤔

**Review by glenn-klaviyo — APPROVED** [review]
looks good to me, left a couple small comments

File: Sources/KlaviyoSwift/StateManagement/KlaviyoState+RequestBuilding.swift (hunk 4 above)
Comment (glenn-klaviyo) [reply Sources/KlaviyoSwift/StateManagement/KlaviyoState+RequestBuilding.swift]: although, this looks to be an existing function so ok if you don't want to rename this.

File: Sources/KlaviyoCore/RequestEnqueuer.swift [hunk 5]
```diff
@@ -59,9 +59,13 @@ public enum RequestEnqueuer {
...
+    public static func enqueueProfile(payload: CreateProfilePayload) {
```
[the remark this answers is not in the document]
Comment (belleklaviyo) [reply Sources/KlaviyoCore/RequestEnqueuer.swift:68]: no existing client builds, this isn't a real bug

File: Sources/KlaviyoCore/QueueStore.swift [hunk 6]
```diff
@@ -129,6 +129,20 @@ public final class QueueStore {
...
 
+    /// Atomically snapshots and clears the pending queue, returning the drained requests.
+    /// Snapshot + clear happen under a single `queueLock` acquisition so a concurrent `enqueue`
+    /// cannot interleave between read and clear. Parity with the flush loop's former
+    /// `requestsInFlight.append(contentsOf: queue); queue.removeAll()`.
+    public func drainAll(persist: PersistPolicy = .debounced) -> [KlaviyoRequest] {
+        let drained = queueLock.withLock { () -> [KlaviyoRequest] in
+            let out = hydrated()
+            queue = []
+            return out
```
[the remark this answers is not in the document]
Comment (belleklaviyo) [reply Sources/KlaviyoCore/QueueStore.swift]: Fixed in a35e5638. Renamed `out` to `drainedRequests`.

-Claude

File: Sources/KlaviyoSwift/StateManagement/KlaviyoState+RequestBuilding.swift [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+
+    mutating func enqueueProfileRequest(apiKey: String, anonymousId: String) {
+        let payload = RequestFactory.profilePayload(
+            identity: requestIdentity(apiKey: apiKey, anonymousId: anonymousId)
+        )
+        let updatedPayload = updateRequestAndStateWithPendingProfile(profile: payload)
+        enqueueRequest(request: RequestFactory.profileRequest(apiKey: apiKey, payload: updatedPayload))
+    }
+
+    mutating func updateRequestAndStateWithPendingProfile(profile: CreateProfilePayload) -> CreateProfilePayload {
```
[the remark this answers is not in the document]
Comment (belleklaviyo) [reply Sources/KlaviyoSwift/StateManagement/KlaviyoState+RequestBuilding.swift:76]: Fixed in a35e5638. Split `updateRequestAndStateWithPendingProfile` into `mergePendingAttributes(from:into:)` and `mergedLocation(from:into:)` helpers; the coordinator is now ~10 lines and both helpers are under the complexity/body-length thresholds.

-Claude

File: Sources/KlaviyoSwift/StateManagement/KlaviyoState+RequestBuilding.swift [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+            location.longitude = location.longitude ?? longitude
+        }
+
+        attributes.location = location
+        self.pendingProfile = nil
+
+        return .init(data: .init(attributes: attributes))
+    }
+
+    /// Validates the requested channels against the profile's identifiers and builds the create-subscription
```
[the remark this answers is not in the document]
Comment (belleklaviyo) [reply Sources/KlaviyoSwift/StateManagement/KlaviyoState+RequestBuilding.swift]: Fixed in a35e5638. Wrapped the doc comment to <= 110 chars.

-Claude

File: Tests/KlaviyoSwiftTests/LegacyStateTests.swift [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+@testable import KlaviyoCore
+@testable import KlaviyoSwift
+import XCTest
+
+final class LegacyStateTests: XCTestCase {
+    func testDecodesNewFormatNestedIdentity() throws {
+        let json = Data("""
+        {"apiKey":"k","identity":{"email":"a@b.com","anonymousId":"anon"},"queue":[]}
+        """.utf8)
+        let decoded = try JSONDecoder().decode(LegacyState.self, from: json)
```
[the remark this answers is not in the document]
Comment (belleklaviyo) [reply Tests/KlaviyoSwiftTests/LegacyStateTests.swift]: Fixed in a35e5638. Extracted a shared `decodeLegacyState(_:)` helper and routed all three tests through it.

-Claude

File: Tests/KlaviyoCoreTests/QueueStoreTests.swift [hunk 10]
```diff
@@ -382,6 +393,65 @@ final class QueueStoreTests: XCTestCase {
...
     }
 
+    // MARK: - drainAll
+
+    func testDrainAllReturnsAndClearsQueue() {
+        let disk = SpyDiskIO([request("a"), request("b")])
+        let scheduler = ManualPersistScheduler()
+        let store = makeStore(diskIO: disk, scheduler: scheduler)
+
+        let drained = store.drainAll()
```
[the remark this answers is not in the document]
Comment (belleklaviyo) [reply Tests/KlaviyoCoreTests/QueueStoreTests.swift:403]: Acknowledged — skipping. The three-line SpyDiskIO + ManualPersistScheduler + makeStore construction is the file-wide convention (~30 tests use it) and the meaningful setup is already extracted into `makeStore`. Adding a tuple-returning helper for only the 4 drain tests would break consistency with the rest of the suite for little gain.

-Claude
